### ISSUE TITLE
Speaker function silences Samsung phones 

### ISSUE THREAD

**Issue by tevas8** [BelledonneCommunications/linphone-android#1531]
Whenever I am in a call from a Samsung phone I expect that tapping on the Speaker icon will allow me to hear the other party through the speaker.
However, when I tap that icon I stop hearing the other party. 
Tapping the speaker again resumes the audio normally (through the earpiece)

There is a way to make it work, which is to touch Samsung phone's buttons for Volume Up or Volume Down (it does not matter which one).
Once you do that, the Speaker will work as expected during that call. (any subsequent calls will display the same bug)

I could not reproduce this on OnePlus phones.
It does happens 100% of the time for Samsung S9 and S10.

**To Reproduce**
1. Login to Linphone on a Samsung S10
2. Dial a number you can pick up
3. Pick up that call
4. Within Linphone tap the Speaker icon
5. Talk on the other side of the line and notice no audio comes through
6. Touch the Volume Up or Volume down button on the left side of the Samsung phone
7. Talk on the other side of the line and notice how the Speaker function works now

**Expected behavior**
Speaker function should always work by simply turning it on without the need of touching any physical device button.

**Please complete the following information**
 - Device: Samsung 10
 - OS: Android 11
 - Version of the App 4.5.4
 - Version of the SDK 5.0.38

**SDK logs**
https://www.linphone.org:444//tmp/61717d4b8fe74_9afc228ed03a8a2af4e3.gz

**Screenshots**

https://user-images.githubusercontent.com/29778372/138303566-23d11c15-0763-4115-a5d8-cf93a2fcbfe8.mp4

**Comment by jeremynorman89** [comment]
Were you able to reproduce this?

**Comment by Viish** [comment]
Sorry but no, I don't have the issue with our Samsung S10 running Android 11.
I even updated it to the latest version (November 1st 2021 security update), switch from earpiece to speaker and vice-versa works fine.

**Comment by driesken** [comment]
I'm also having this issue on a Samsung S10e when using Aaudio. When using OpenSLES, it works perfect. Are we supposed to use Aaudio in latest releases? Or is OpenSLES still preferred?

**Comment by driesken** [comment]
I'm also having this issue on a Samsung S10e when using Aaudio. When using OpenSLES, it works perfect. Are we supposed to use Aaudio in latest releases? Or is OpenSLES still preferred?

**Comment by Viish** [comment]
OpenSLES has been deprecated by Google, so AAudio is now preferred. 
However on many Samsung devices there are issues, even for Oboe team...
Have you / can you try the 4.6 beta on the Play Store? As I said I don't reproduce it with our Samsung S10 here at the office.

**Comment by driesken** [comment]
Thanks for the info.

I've just enrolled into the beta and made 2 calls (first one with Aaudio where speaker fails, second one with OpenSLES). I've uploaded the logfile.

PS: I had to exit Linphone after initial set-up to get the speaker working within OpenSLES. Afterwards, I could easily switch between them to reproduce the issue.

https://www.linphone.org:444//tmp/61a6711e85958_774b5bb655c6de93c98a.gz
